community/cypher/cypher/src/main/scala/org/neo4j/cypher/internal/spi/v3_3/TransactionBoundPlanContext.scala

```diff
@@ -0,0 +1,196 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.cypher.internal.spi.v3_3
+
+import java.util.Optional
+
+import org.neo4j.cypher.MissingIndexException
+import org.neo4j.cypher.internal.compiler.v3_3.IndexDescriptor
+import org.neo4j.cypher.internal.compiler.v3_3.spi._
+import org.neo4j.cypher.internal.frontend.v3_3.phases.InternalNotificationLogger
+import org.neo4j.cypher.internal.frontend.v3_3.symbols.CypherType
+import org.neo4j.cypher.internal.frontend.v3_3.{CypherExecutionException, symbols}
+import org.neo4j.cypher.internal.runtime.interpreted.LastCommittedTxIdProvider
+import org.neo4j.cypher.internal.v3_3.logical.plans._
+import org.neo4j.internal.kernel.api.exceptions.KernelException
+import org.neo4j.kernel.api.index.InternalIndexState
+import org.neo4j.kernel.api.proc.Neo4jTypes.AnyType
+import org.neo4j.kernel.api.proc.{Neo4jTypes, QualifiedName => KernelQualifiedName}
+import org.neo4j.kernel.api.schema.SchemaDescriptorFactory
+import org.neo4j.kernel.api.schema.index.{IndexDescriptor => KernelIndexDescriptor}
+import org.neo4j.kernel.impl.proc.Neo4jValue
+import org.neo4j.procedure.Mode
+
+import scala.collection.JavaConverters._
+
+class TransactionBoundPlanContext(tc: TransactionalContextWrapper, logger: InternalNotificationLogger)
```

Comment:
To get rid of TransactionalContextWrapper, we should pass in instead a `readOperationsSupplier` of type `() => ReadOperations`